### PR TITLE
Static method to create AbstractContext subclass

### PR DESCRIPTION
**Pull request by sebastianbaginski**
To reduce code duplication in other project.
@davidhesselbom can you review ?

### CODE REVIEW

**Comment by bettinaselig** [comment]
We only use "peer review" and "final review" for pull requests. The other labels are for issues from now on.

**Comment by sebastianbaginski** [comment]
ok

File: source/draw/AbstractContext.ooc [hunk 1]
```diff
@@ -36,4 +36,6 @@ AbstractContext: abstract class {
 	alignWidth: virtual func (width: Int, align := AlignWidth Nearest) -> Int { width }
 	update: abstract func
 	isAligned: virtual func (width: Int) -> Bool { true }
+	_createContext := static func -> This { CpuContext new() }
```
Comment (davidhesselbom) [inline source/draw/AbstractContext.ooc]: `_createContext := static func`? This syntax is unfamiliar...

Comment (sebastianbaginski) [inline source/draw/AbstractContext.ooc]: It is just like `defaultValue := static 42`

Comment (davidhesselbom) [inline source/draw/AbstractContext.ooc]: Okay, so it's like `_createContext ::= This CpuContext new()`?

Comment (davidhesselbom) [inline source/draw/AbstractContext.ooc]: Wait, no, that's a property...

Comment (davidhesselbom) [inline source/draw/AbstractContext.ooc]: But how is it different from `_createContext: static func -> This { CpuContext new() }`?

Comment (sebastianbaginski) [inline source/draw/AbstractContext.ooc]: I wanted `_createContext` to be a static data member, Closure which you can reassign later. I think in your example above one cannot reassign anything to `_createContext`.

Comment (davidhesselbom) [inline source/draw/AbstractContext.ooc]: Oh, right. Got it. We don't use reassignable func members a lot, so I wasn't even sure what this did. Thanks.

Comment (sebastianbaginski) [inline source/draw/AbstractContext.ooc]: Sure.
This design is actually Simon's idea, but unfortunately I could not have done it like
```ooc
_createContext := CpuContext new
```
as he originally suggested. I needed to wrap it in a closure here.

Comment (davidhesselbom) [inline source/draw/AbstractContext.ooc]: Looks cool though!

**Comment by marcusnaslund** [comment]
@davidhesselbom - are you satisfied with this enough to set the green flag?

**Comment by davidhesselbom** [comment]
Looks good enough to me, but I haven't tested it.

**Comment by sebastianbaginski** [comment]
Unfortunately this does not work as it should. I need to remove the flag for now.

**Comment by sebastianbaginski** [comment]
It won't work this way. I'm closing this.

**Pull request closed without merging**